### PR TITLE
Feature/support custom uri segments

### PR DESCRIPTION
**Pull request by piotr-szczygiel**
- created `Forte.EpiResponsivePicture.Blob.BlobImageProvider` out of codebase from `Baaijte.Optimizely.ImageSharp.Web.Providers.BlobImageProvider`
- new class has a possibility to set custom URI segments (in original class list of segments was hardcoded)

### CODE REVIEW

File: EpiResponsivePicture/Blob/BlobImageProvider.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        "/contentassets", "/globalassets", "/siteassets"
+    };
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="BlobImageProvider"/> class.
+    /// </summary>
+    /// <param name="environment">The environment used by this middleware.</param>
+    /// <param name="formatUtilities">Contains various format helper methods based on the current configuration.</param>
+    public BlobImageProvider(IWebHostEnvironment environment, FormatUtilities formatUtilities)
```
Comment (tomaszmadeyski) [inline EpiResponsivePicture/Blob/BlobImageProvider.cs]: `environment` seems not to be used

File: EpiResponsivePicture/Blob/BlobImageProvider.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using SixLabors.ImageSharp.Web;
+using SixLabors.ImageSharp.Web.Providers;
+using SixLabors.ImageSharp.Web.Resolvers;
+
+namespace Forte.EpiResponsivePicture.Blob;
+
+public class BlobImageProvider : IImageProvider
```
Comment (tomaszmadeyski) [inline EpiResponsivePicture/Blob/BlobImageProvider.cs]: I wonder if it's not better to extend `Baaijte.Optimizely.ImageSharp.Web.Providers.BlobImageProvider` and override `Match` method (and of course add all stuff like AddSegments etc)? This class is tightly coupled with Baajite implementation anyway (due to `BlobImageResolver` usage) so we can avoid rewriting GetAsync, ProcessingBehavior etc.
What do you think?

Comment (piotr-szczygiel) [reply EpiResponsivePicture/Blob/BlobImageProvider.cs]: You're right. But I changed it even better I think. Using original `BlobImageResolver` from `Baaijte.Optimizely.ImageSharp.Web.Providers` - figured out that `Match` method has public setter, so I can assign custom functionality into it.

File: EpiResponsivePicture/Extensions/ApplicationBuilderExtensions.cs [hunk 3]
```diff
@@ -13,12 +18,16 @@ public static class ApplicationBuilderExtensions
...
-    public static IApplicationBuilder UseForteEpiResponsivePicture(this IApplicationBuilder app, Func<IApplicationBuilder, IApplicationBuilder> providerRegistration = null)
+    public static IApplicationBuilder UseForteEpiResponsivePicture(this IApplicationBuilder app, 
+        Func<IApplicationBuilder, IApplicationBuilder> providerRegistration = null)
     {
-
         providerRegistration ??= SixLabors.ImageSharp.Web.DependencyInjection.ApplicationBuilderExtensions.UseImageSharp;
-        
         providerRegistration(app);
+
+        var imageProvider = app.ApplicationServices.GetServices<IImageProvider>()
```
Comment (tomaszmadeyski) [inline EpiResponsivePicture/Extensions/ApplicationBuilderExtensions.cs:27]: how about `var imageProvider = app.ApplicationServices.GetService<BlobImageProvider>();` ?

File: EpiResponsivePicture/Extensions/ServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -82,9 +82,11 @@ private static void ConfigureModule(IServiceCollection services, EpiResponsivePi
         services
             .AddOptions<EpiResponsivePicturesOptions>()
             .Configure(o => {
-                o.ImageResizerCompatibilityEnabled = options?.ImageResizerCompatibilityEnabled ?? false; 
+                o.ImageResizerCompatibilityEnabled = options?.ImageResizerCompatibilityEnabled ?? false;
+                o.AdditionalSegments = options?.AdditionalSegments;
             });
 
+        services.AddScoped<IBlobSegmentsProvider, BlobCustomSegmentsProvider>();
```
Comment (tomaszmadeyski) [inline EpiResponsivePicture/Extensions/ServiceCollectionExtensions.cs]: as far as I can see BlobImageProvider is registered as singleton, so maybe this registration can also be done with Singleton?

File: EpiResponsivePicture/Extensions/ApplicationBuilderExtensions.cs (hunk 3 above)
Comment (piotr-szczygiel) [reply EpiResponsivePicture/Extensions/ApplicationBuilderExtensions.cs:27]: That won't work, as `BlobImageProvider` is not registered directly as class instance.

Comment (tomaszmadeyski) [reply EpiResponsivePicture/Extensions/ApplicationBuilderExtensions.cs:27]: ok, so one last thing - maybe it's better to do FirstOrDefault() and throw exception saying that there's no service registered for BlobImageProvider? Default exception (sequence contains no elements) might be hard to debug

Comment (piotr-szczygiel) [reply EpiResponsivePicture/Extensions/ApplicationBuilderExtensions.cs:27]: Fixed as suggested.

**Review by tomaszmadeyski — APPROVED** [review]

**Pull request merged**